src/lattice-core/connected/web_socket.cr

```diff
@@ -241,7 +250,13 @@ module Lattice::Connected
 
     def self.on_message(message, socket)
 
-      payload = validate_payload(message, socket)
+      unless (payload = validate_payload(message, socket))
+        puts "No payload for #{message}"
+        return
+      else
+        payload = payload.as(ValidatedPayload)
```

Comment:
Because the validate_payload method can now return a nil (invalid parsing), the payload now has to be cast into the non-nil variant.  It's a messy type, so it's been `aliased` at the beginning of the class.